### ISSUE TITLE
Create sub-category 

### ISSUE THREAD

**Issue by sopheaouk** [grokability/snipe-it#8366]
Dear Snipe-IT,

I have checked with snipe-IT sales team about a possibility to add a sub-category for an asset. This feature is really useful for many org in term of Asset management. 

For an example: I have many network devices , Router --> Cisco and Huawie, Firewall --> Palo Alto, Cisco ASA, Switch --> Dell , Cisco...and so on.

Feature needs sample for Categories and sub categories:

1--Network Devices
   1.1--Router
       1.1.1--Cisco 
   1.2--Switch
      1.2.1--Cisco --> Model (Cisco 2960, 3550...)
   1.3--Firewall 

A waiting for a new updates ...

From a Snipe-IT lover...
Best...
 

**Server (please complete the following information):**
 - Snipe-IT Version 4.9.4
 - OS: CentOS]
 - Web Server: [e.g. Apache, IIS]
 - PHP Version 

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by snipe** [comment]
That's unfortunately a lot less trivial than it would seem at first glance, as we end up with the need to show all assets within a category including all of the subcategories, which are recursive functions and can put a lot of strain on the database. We can take a look at this after v5 is out, but it's not on the short-term roadmap.

**Comment by sopheaouk** [comment]
> That's unfortunately a lot less trivial than it would seem at first glance, as we end up with the need to show all assets within a category including all of the subcategories, which are recursive functions and can put a lot of strain on the database. We can take a look at this after v5 is out, but it's not on the short-term roadmap.

Thank for your quick response ...anyway, v5 we are still waiting ...

**Comment by snipe** [comment]
As I mentioned elsewhere, we're working on an RC for the end of this week, and will hopefully have v5 out mid-Sept.

**Comment by lheuveldop** [comment]
I would still find the feature cool, so please don't close it if it's not impossible

**Comment by jessetechno** [comment]
This is by far the BIGGEST thing missing from Snipe-IT.  This software is amazing and I love using it but... This is a huge need.

**Comment by shakilak2005** [comment]
Hi, kindly add this feature of the sub-category. This is an important feature and can help a lot to the users of the software.

**Comment by measyberger** [comment]
Dear Snipe-IT contributers, 
today it came to my mind that sub-categories could possibly help us with the organization-wide use of Snipe-IT at Ars Electronica. I think we could make really good use of a feature like this because we have 35 categories and it might bring tidiness into them (users might select the wrong category if it's so confusing because there are so many) and it might also allow us to create even more categories without it becoming cluttered. 
Unfortunately, because I'm not a software developer, I can't contribute much except my ideas. And since this idea is still quite young, it doesn't have to be the best either. :-D 
Thank you for your great work!

**Comment by suka97** [comment]
I agree that such a feature would be greatly needed for managing electronic components and such

**Comment by snipe** [comment]
Again though, I think that using clever category names solves 95% of the use-cases here, and spares us the database overhead/performance of potentially recursive queries. This seems like an administrative solve, not a technological one.
